cloud/audit/v1/LogEntryData.ts

```diff
@@ -283,7 +283,9 @@ export interface ServiceMetadata {
   /**
    * Unordered map of dynamically typed values.
    */
-  fields?: {[key: string]: {[key: string]: any}};
+  fields?: {
+    [key: string]: any[] | boolean | number | {[key: string]: any} | string;
```

Comment:
Is the recursive map definition correct? If the value for a struct field is another struct, that should basically be the same type of value as the top level here. I don't know enough Typescript to know whether this is an issue or not.

Reply:
Yes, from what I can read.

https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#google.protobuf.Struct

Key is string.

Value (here) is list_value, bool_value, number_value, struct_value, string_value. null_value is represented with the ? in the property.

Reply:
So is "any" just "any JSON object" (but not other JSON values)? The problem is I understand the proto well, but not the Typescript :)

Value is expected to represent any JSON value (string, number, array, object, bool) and Struct is expected to represent any JSON object... I just can't tell for sure whether that's what the Typescript represents.

Reply:
any in typescript means anything, including non JSON values like undefined, null, classes, etc.

Reply:
There is no basic JSON type.

See https://github.com/microsoft/TypeScript/issues/1897